src/containers/Tenant/Query/QueryEditor/QueryEditor.tsx

```diff
@@ -101,6 +103,7 @@ function QueryEditor(props: QueryEditorProps) {
     const [resultType, setResultType] = React.useState(RESULT_TYPES.EXECUTE);
     const [isResultLoaded, setIsResultLoaded] = React.useState(false);
     const [querySettings] = useQueryExecutionSettings();
+    const [currentQueryId, setCurrentQueryId] = React.useState<string | null>(null);
```

Comment:
Maybe we can get current query id from `executeQueryResult` or `explainQueryResult`, something like this
```ts
let currentQueryId;
if (executeQueryResult.isLoading) {
    currentQueryId = executeQueryResult.originalArgs.queryId;
} else if (explainQueryResult.isLoading) {
    currentQueryId = explainQueryResult.originalArgs.queryId;
}
```